### ISSUE TITLE
[Bug]: mpool leak in remote run

### ISSUE THREAD

**Issue by nnsgmsone** [matrixorigin/matrixone#14263]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1-dev

### Commit ID

e420387693a219554840878d9872cfc192738b89

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

remoterun使用过一个mpool后并不会移除这个mpool，这个会导致这个mpool的泄漏

### Steps to Reproduce

```Markdown
1. 多cn
2. run tpch
```

**Comment by nnsgmsone** [comment]
fixed by #14264

**Comment by aressu1985** [comment]
fixed
